code/game/objects/items/devices/illegalradio.dm

```diff
@@ -0,0 +1,177 @@
+/obj/item/device/illegalradio
+	icon = 'icons/obj/radio.dmi'
+	name = "black market uplink"
+	suffix = "\[3\]"
+	icon_state = "illegalradio"
+	item_state = "illegalradio"
+	desc = "A modified radio with a link to the black market. Use with caution."
+
+	siemens_coefficient = 1
+	slot_flags = SLOT_BELT
+	throw_speed = 2
+	throw_range = 9
+	w_class = W_CLASS_SMALL
+	starting_materials = list(MAT_IRON = 75, MAT_GLASS = 25)
+	w_type = RECYK_ELECTRONIC
+	melt_temperature = MELTPOINT_PLASTIC
+	
+	var/money_stored		        // Money placed in the buffer
+	// List of items not to shove in their hands.
+	var/list/purchase_log = list()
+	var/show_description = null
+	var/active = 0
+	var/job = null
+	
+/obj/item/device/illegalradio/New()
+	..()
+	if(ticker)
```

Comment:
where does `ticker` come from?

Reply:
and what does it do

Reply:
It's a var that keeps references to the gamemode, the list of minds, and a few other things that depend on the game being initialised.

Reply:
why is it relevant for use, i know this apparently is a null check for ticker.mode but i also dont understand that logic